### ISSUE TITLE
[Feature request] Enable copying current position in the "Go To" dialog

### ISSUE THREAD

**Issue by bugpwr** [notepad-plus-plus/notepad-plus-plus#15950]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

adding a "Copy" button in the Go To dialog. The button will copy either the current line number or the file offset to the clipboard, depending on the selected mode (Go To Line or Go To Offset).

(on an anecdotal note, while manually copying a file offset in a huge gcode manually I made a mistake crashing my cnc machine head)

### Describe the solution you'd like.

a neat "copy" button in the goto dialog. I will commit a fix in a moment

### Debug Information

```shell
Notepad++ v8.7.4   (64-bit)
Build time : Dec  4 2024 - 23:50:05
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
Periodic Backup : ON
Placeholders : OFF
DirectWrite : OFF
Multi-instance Mode : monoInst
File Status Auto-Detection : cdEnabledNew (for current file/tab only)
Dark Mode : ON
OS Name : Windows 11 Home (64-bit)
OS Version : 23H2
OS Build : 22631.4602
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
```

### Anything else?

this is what I intend to commit. 
				case IDC_COPY_OFFSET:
					if (OpenClipboard(_hSelf))
					{
						char clipboardText[256];
						size_t position = 0;
						if (_mode == go2line)
							position = (*_ppEditView)->getCurrentLineNumber() + 1;
						else
							position = (*_ppEditView)->execute(SCI_GETCURRENTPOS);
			
						snprintf(clipboardText, sizeof(clipboardText), "%zu", position);

						EmptyClipboard();
						size_t textLen = strlen(clipboardText) + 1;
						HGLOBAL hGlobal = GlobalAlloc(GMEM_MOVEABLE, textLen);
						if (hGlobal)
						{
							char* pGlobal = static_cast<char*>(GlobalLock(hGlobal));
							if (pGlobal)
							{
								memcpy(pGlobal, clipboardText, textLen);
								GlobalUnlock(hGlobal);
								SetClipboardData(CF_TEXT, hGlobal);
							}
						}
						CloseClipboard();
					}
					return TRUE;

**Comment by alankilborn** [comment]
Is such a thing really useful?

**Comment by bugpwr** [comment]
> Is such a thing really useful?

yes please. otherwise one has to copy a 8-digit number by  hand. Not how computers are supposed to work

**Comment by bugpwr** [comment]
for donho's conveience, this is what it looks like to extract a byte position on an industrial 3d printer to resume after trouble, and if you make a mistake, you will bang into the model potentially destroying equipment. that's why I added it
![image](https://github.com/user-attachments/assets/4472ef04-9cd4-4d86-9e6f-b7fa6931f2b4)

**Comment by alankilborn** [comment]
@bugpwr

What if the _You want to go to:_ box is populated with the _You are here:_ box value when the command that brings up the dialog box is invoked?  Then it is "reasonably" easy to copy the value.

...and the _You are here:_ "row" in the dialog could be eliminated (or...it could stay and be somewhat redundant).

---

Other unrelated notes about the _Go To..._ dialog:

* If you press _Go_ when there is no text in the _You want to go to:_ box, focus is returned to the editor (caret position doesn't change) and the dialog remains open (it is usually closed when _Go_ is pressed -- but see below)
* It is possible for "here" to be greater than the "can't go further than" value, e.g.:
![image](https://github.com/user-attachments/assets/ac5054c0-cd9b-489d-9dcc-666892025605)
* It might be nicer if there was a way to _Go_ but have the _Goto..._ dialog remain open (It's already modeless, so...); the addition of this would also make it better for when you specify an offset that is in the middle of a utf-8 multibyte character (and the _Go_ button moves you to a different position value that you specify, to avoid being in the middle of that character)

**Comment by bugpwr** [comment]
"What if the You want to go to: box is populated with the You are here: box value when the command that brings up the dialog box is invoked? Then it is "reasonably" easy to copy the value." -- good idea, this is how it is implemented in regular notepad. I did something else (see pic) and this got 'rejected' for now, please upvote

![image](https://github.com/user-attachments/assets/7f04b908-5099-4a9a-8387-2ef79036130a)

**Comment by donho** [comment]
I'll see what I can do about it (No extra button).

**Comment by bugpwr** [comment]
well, I can change the code of course to make it a simple edit box

**Comment by bugpwr** [comment]
please merge at your convenience, thanks.
